### PR TITLE
[FEATURE] Detailed filters on dates 3865

### PR DESCRIPTION
**Pull request by hschink**
Implements [ticket 3865](https://www.openproject.org/work_packages/3865)

### CODE REVIEW

File: app/models/work_package.rb [hunk 1]
```diff
@@ -78,6 +78,10 @@ class WorkPackage < ActiveRecord::Base
     {:conditions => {:project_id => projects}}
   }
 
+  scope :changed_since, lambda { |changed_since|
+    changed_since ? { conditions: ["updated_at >= ?", changed_since] } : nil
```
Comment (hschink) [inline app/models/work_package.rb]: `conditions` should become `where`

File: app/controllers/api/v2/planning_elements_controller.rb [hunk 2]
```diff
@@ -278,6 +280,14 @@ def rewire_ancestors
 
 
       end
+
+      private
+
+      def parse_changed_since
+        if params[:changed_since]
```
Comment (hschink) [inline app/controllers/api/v2/planning_elements_controller.rb]: This may be superfluous as `nil.to_i` becomes `0` which resolves to the 1. January 1970.

**Comment by hschink** [comment]
Approved by Patrick

File: app/models/work_package.rb [hunk 3]
```diff
@@ -78,6 +78,10 @@ class WorkPackage < ActiveRecord::Base
     {:conditions => {:project_id => projects}}
   }
 
+  scope :changed_since, lambda { |changed_since|
+    changed_since ? where(["updated_at >= ?", changed_since]) : nil
```
Comment (GarstgerUnhold) [inline app/models/work_package.rb]: The scope breaks at this line, I think you need to provide the `WorkPackage.table_name` for updated_at just as in the "open" scope below. At least, `features/step_definitions/api_teps.rb:102` fails on Travis and it looks like the cause might be this scope.

**Pull request merged**